### PR TITLE
dot/rpc: implement RPC methods chain_getHeader and chain_getBlock

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes
- Implement RPC methods and tests for chain_getHeader
- Implement RPC methods and test for chain_getBlock

## Tests:
Unit tests:
```
go test ./dot/rpc/... -v
```
Interactive testing, start node:
```
make gossamer
./bin/gossamer --key alice --roles 4 --rpc --rpcmods chain,system
```
Interact with node via RPC, call chain_getHeader for latest header
```
curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getHeader", "params": []}' http://localhost:8545/
```
Interact with node via RPC, call chain_getHeader for hash 
```
curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getHeader", "params": ["0x64be6cb9aaa36571309cb9f9c3dcf08197c8d1f80baff58e80bd1bb188d909fe"]}' http://localhost:8545/
```

Similar for calls to chain_getBlock 

### Issues:
This code introduces bug described in issue #744

### CODE REVIEW

File: dot/rpc/modules/chain.go [hunk 1]
```diff
@@ -53,21 +69,73 @@ func NewChainModule(api BlockAPI) *ChainModule {
...
+	return nil
 }
 
 // GetBlockHash isn't implemented properly yet.
-func (cm *ChainModule) GetBlockHash(r *http.Request, req *ChainBlockNumberRequest, res *ChainHashResponse) {
+// TODO finish this
+func (cm *ChainModule) GetBlockHash(r *http.Request, req *ChainBlockNumberRequest, res *ChainHashResponse) error {
+	// TODO get values from req
+	fmt.Printf("blockHash %s\n", *req)
+	fmt.Printf("type %v\n", reflect.TypeOf(req))
```
Comment (noot) [inline dot/rpc/modules/chain.go:107]: remove this?

Comment (noot) [reply dot/rpc/modules/chain.go:107]: unless you plan to finish it in this PR

File: dot/rpc/modules/chain_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		t.Fatal(err)
+	}
+	err = stateSrvc.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return stateSrvc
+}
+
+func TestChainGetHeader_Gensis(t *testing.T) {
```
Comment (noot) [inline dot/rpc/modules/chain_test.go:37]: genesis*

File: dot/rpc/modules/chain_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	chain := newChainService(t)
+	svc := NewChainModule(chain.Block)
+
+	res := &ChainBlockHeaderResponse{}
+	req := ChainHashRequest("zz")
+	err := svc.GetHeader(nil, &req, res)
+	require.EqualError(t, err, "could not byteify non 0x prefixed string")
+}
+
+func TestChainGetBlock_Gensis(t *testing.T) {
```
Comment (noot) [inline dot/rpc/modules/chain_test.go:93]: genesis*

**Comment by edwardmack** [comment]
I think this is ready now...

**Pull request merged**